src/connection/builders/builder_native.rs

```diff
@@ -0,0 +1,301 @@
+        let mut self_result = Self {
+            _marker_linkage: PhantomData,
+            _marker_conntype: PhantomData,
+            conn_conf: Default::default(),
+            charset: charset::UTF_8,
+            lib_path: None,
+        };
+
+        self_result.conn_conf.dialect = Dialect::D3;
+        self_result.conn_conf.stmt_cache_size = 20;
+        self_result.conn_conf.attachment_conf.remote = None;
+        self_result.conn_conf.attachment_conf.user = "SYSDBA".to_string();
+        self_result.conn_conf.attachment_conf.db_name = "test.fdb".to_string();
+
+        self_result
+    }
+}
+
+//can only use these methods on a remote builder
+impl<A> NativeConnectionBuilder<A, Remote> {
+    //private helper accessor for the Option<RemoteConfig> buried inside
+    //the configuration
+    fn get_initialized_remote(&mut self) -> &mut RemoteConfig {
+        self.conn_conf
+            .attachment_conf
+            .remote
+            .get_or_insert(Default::default())
+    }
+
+    /// Hostname or IP address of the server. Default: localhost
+    pub fn host<S: Into<String>>(&mut self, host: S) -> &mut Self {
+        self.get_initialized_remote().host = host.into();
+        self
+    }
+
+    /// TCP Port of the server. Default: 3050
+    pub fn port(&mut self, port: u16) -> &mut Self {
+        self.get_initialized_remote().port = port;
+        self
+    }
+
+    /// Password. Default: masterkey
+    pub fn pass<S: Into<String>>(&mut self, pass: S) -> &mut Self {
+        self.get_initialized_remote().pass = pass.into();
+        self
+    }
+}
+
+impl<A> NativeConnectionBuilder<A, ConnTypeNotConfigured> {
+    /// Configure the native client for remote connections.
+    /// This will allow configuration via the 'host', 'port' and 'pass' methods.
+    pub fn as_remote(mut self) -> NativeConnectionBuilder<A, Remote> {
+        let mut remote: RemoteConfig = Default::default();
+        remote.host = "localhost".to_string();
+        remote.port = 3050;
+        remote.pass = "masterkey".to_string();
+        self.conn_conf.attachment_conf.remote = Some(remote);
+        self.safe_transmute()
+    }
+
+    //does nothing since the embedded config is common to both connection types
+    /// Configure the native client for embedded connections.
+    /// There is no 'host', 'port' or 'pass' to configure on the result of this
+    /// method and attempts to call those methods will result in a
+    /// compile error.
+    ///
+    /// Note that the embedded builder is only tested for firebird >=3.0.
+    /// If the embedded connection fails, the client dll may attempt to use
+    /// other means of connection automatically, such as XNET or localhost.
+    ///
+    /// On firebird 3.0 and above this may be restricted via the `Providers`
+    /// config parameter of `firebird.conf` see official firebird documentation
+    /// for more information.
+    pub fn as_embedded(self) -> NativeConnectionBuilder<A, Embedded> {
```

Comment:
Clippy warning.
Consider changing to `into_embedded` or suppressing the warning using `#[allow(clippy::wrong_self_convention)]`.